### PR TITLE
chore : add FEN to CFBoard conversion

### LINKED ISSUE #28: [ALGO] Implement FEN
Implement:
- [x] `fromFEN`
- [x] `toFEN`

Note that @souhhcong will need this functionality soon, and if @Yoshimi-Theophile have to do something else, you can assign it to me (or someone else).

### PR DESCRIPTION
**Pull request by souhhcong**
Implemented the conversion:
- Add a function to split a string so the code is more readable
- Convert each field of the FEN string to the corresponding CFBoard representation (No 50-move draw rule implementation yet because it's not in priority currently)
- Add some semi-colon and round brackets to Yoshi's code so I could compile and test.

Closes #28

### CODE REVIEW

File: lib/board_implementation/CFBoard.cpp [hunk 1]
```diff
@@ -1,13 +1,94 @@
...
+    pawnBoard = 0LL;
+	kingBoard = 0LL;
+	queenBoard = 0LL;
+	rookBoard = 0LL;
+	knightBoard = 0LL;
+	bishopBoard = 0LL;
+	blackBoard = 0LL;
+	whiteBoard = 0LL;
+	turn = 0;
+    castleBools = 0;
```
Comment (plumsirawit) [inline lib/board_implementation/CFBoard.cpp:21]: Inconsistent indentation.

**Review by plumsirawit — CHANGES_REQUESTED** [review]
Please check indentation, thanks. (I think we're using one tab for everything.)

File: lib/board_implementation/CFBoard.cpp (hunk 1 above)
Comment (souhhcong) [reply lib/board_implementation/CFBoard.cpp]: Changed all 4 spaces to tabs in the next commit.

**Review by plumsirawit — APPROVED** [review]
Looks good!

**Pull request merged**